File: Snake-Case-To-Camel-Case.cs
//A simple challenge to change a string written in "snake-case" to Camel-Case.
//I did this exercise already in Typescript but wanted to see how it would differ when using strong typing and the array restrictions of c#

string SnakeCaseToCamelCase(string value)
		{
			string result = "";
			string? firstLetter;
			string[] temp = value.Split("_");
			
			if (value.Length == 0)
				return "";
			else if (value.Length == 1)
				return value.ToLower();
			else
			{
				foreach (var el in temp)
				{
					result += el.Substring(0,1).ToUpper() + el.Substring(1, el.Length - 1);
				}
					firstLetter = result.Substring(0,1).ToLower();
					result = firstLetter + result.Substring(1, result.Length - 1);
				return result;
			}
		}
		
		Console.WriteLine(SnakeCaseToCamelCase("please_print_hello_world"));

File: Snake-Case-To-Camel-Case.ts
//A small challenge to take a "snake-case" string and return the same string but in Camel-case instead.
//Example => this_is_an_example > thisIsAnExample

function snakeCaseToCamelCase(value : string) {
  let arr = [];
  let result = "";
  const temp = value.split("_");
  
  for (const el of temp) {
    arr.push(el.charAt(0).toUpperCase() + el.slice(1).toLowerCase());
  }
  for (let i = 0; i < arr.length; i++) {
    result += arr[i];
  }
  return result.charAt(0).toLowerCase() + result.slice(1).toString();
}

console.log(snakeCaseToCamelCase("hello_world_with_types"));
console.log(snakeCaseToCamelCase("HELLO_WORLD_WITH_TYPES"));
